### PR TITLE
Add docs for GHC-78892

### LINKED ISSUE #61: PsErrLetInPat -> 78892

### PR DESCRIPTION
**Pull request by TauOmicronMu**
Closes #61

### CODE REVIEW

File: message-index/messages/GHC-78892/index.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: let-syntax in pattern
+summary: An attempt has been made to use a let expression whilst pattern matching. 
+severity: error
+introduced: 9.6.1
+---
+
+Pattern matching allows constructors and literals to be used to take apart values, exposing their contents. The `let ... in ...` syntax is used to introduce declarations that are scoped to an expression. See [GHC-04584](https://errors.haskell.org/messages/GHC-78892/) for more information on why expressions aren't permitted in pattern matching.
```
Comment (david-christiansen) [inline message-index/messages/GHC-78892/index.md:8]:
```suggestion
Pattern matching allows constructors and literals to be used to take apart values, exposing their contents. Even though patterns resemble expressions, they are not evaluated. The `let ... in ...` syntax is used to introduce declarations that are scoped to an expression, which doesn't make sense as a pattern without evaluation. See [GHC-04584](https://errors.haskell.org/messages/GHC-78892/) for more information on why expressions aren't permitted in pattern matching. 
```

File: message-index/messages/GHC-78892/example/index.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+sq :: Int -> Int 
+sq x = x^2 
+
+f :: Int -> Int -> Bool 
+f n (sq n) = True
+f n _ = False 
+
+``` 
+
+The above code would throw a parse error in the pattern [GHC-07626](https://errors.haskell.org/messages/GHC-07626/).
```
Comment (david-christiansen) [inline message-index/messages/GHC-78892/example/index.md:29]: :+1:

**Review by david-christiansen — COMMENTED** [review]
It's OK as is, but I suggested a minor tweak to the wording

**Comment by david-christiansen** [comment]
Thanks!

File: message-index/messages/GHC-78892/index.md (hunk 1 above)
Comment (TauOmicronMu) [reply message-index/messages/GHC-78892/index.md]: Changed :)

**Review by david-christiansen — APPROVED** [review]

**Comment by david-christiansen** [comment]
Thank you again :-)

**Pull request merged**